test/file_test.rb

```diff
@@ -107,13 +114,14 @@ def test_open_buffer_with_stringio
   def test_close_buffer_with_stringio
     string_io = StringIO.new File.read('test/data/rubycode.zip')
     zf = ::Zip::File.open_buffer string_io
-    assert(zf.close || true) # Poor man's refute_raises
+    assert_nil zf.close
   end
 
   def test_close_buffer_with_io
     f = File.open('test/data/rubycode.zip')
```

Comment:
Note: this is similar (but not identical) to the test cases in https://github.com/rubyzip/rubyzip/issues/146 and https://github.com/rubyzip/rubyzip/issues/192, so they can probably be closed once we sort this out.